### PR TITLE
Fix ASDF issues

### PR DESCRIPTION
**Pull request by Cadair**
This is a follow on to #7585 which fixes the tests which should have been fixed in that PR.

### CODE REVIEW

File: sunpy/tests/figure_hashes_mpl_382_ft_261_astropy_600_animators_111.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
   "sunpy.map.tests.test_compositemap.test_plot_composite_map_colors": "4e6310746beab7a9d866aab10d86878aa5797b2f20ae933f8cfdcdf1ce9d461e",
   "sunpy.map.tests.test_compositemap.test_set_alpha_composite_map": "2b2a157405dc296d5c06c74f7d8bc6e03b3cff4512cf9b0e66a773827497f37a",
   "sunpy.map.tests.test_compositemap.test_peek_composite_map": "a21143a11c499d95b3618d2f82a6d6e8a3cbad6291ab52b20b218b3f4a03c45b",
-  "sunpy.map.tests.test_map_factory.test_map_jp2_HMI": "0b92d617cae5712f3c9a131f2712e74e38cacddb16902a19ea26db424e40d1d2",
+  "sunpy.map.tests.test_map_factory.test_map_jp2_HMI": "3b731f25ab5777f187def8296fbed88d27809867405e0e97d5e808ca1bb4b42c",
```
Comment (Cadair) [inline sunpy/tests/figure_hashes_mpl_382_ft_261_astropy_600_animators_111.json:11]: I assume this change was in the commit I reverted and just shouldn't have been?!

Comment (nabobalis) [reply sunpy/tests/figure_hashes_mpl_382_ft_261_astropy_600_animators_111.json:11]: As long as the figure looks ok and the hash is ok.

Comment (nabobalis) [inline sunpy/tests/figure_hashes_mpl_382_ft_261_astropy_600_animators_111.json:11]:
```suggestion
  "sunpy.map.tests.test_map_factory.test_map_jp2_HMI": "0b92d617cae5712f3c9a131f2712e74e38cacddb16902a19ea26db424e40d1d2",
```

**Review by nabobalis — APPROVED** [review]

**Comment by braingram** [comment]
I think the oldestdeps job is failing because non-vo units weren't supported until asdf-astropy 0.4.0:
https://github.com/astropy/asdf-astropy/blob/main/CHANGES.rst#040-2023-03-20
and the pyproject.toml lists 0.2.0:
https://github.com/sunpy/sunpy/blob/c98a93edbd674fdb75c440e98584b73d14f6b783/pyproject.toml#L56
I think that will also trigger an asdf bump since asdf-astropy 0.4.0 requires asdf 2.13:
https://github.com/astropy/asdf-astropy/blob/0.4.x/pyproject.toml#L20

**Comment by nabobalis** [comment]
oldestdeps fail is real.

**Comment by braingram** [comment]
> oldestdeps fail is real.

jinx

**Comment by nabobalis** [comment]
I am in favour of increasing the min deps to avoid this.

**Pull request merged**